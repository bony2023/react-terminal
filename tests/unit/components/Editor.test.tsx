import * as React from "react";
import { render } from "@testing-library/react"
import ContextProvider from "../../../src/contexts";
import { TerminalContextProvider } from "../../../src/contexts/TerminalContext";
import Editor from "../../../src/components/Editor";

let props: any;
const renderWrapper = () => (
  <ContextProvider>
    <TerminalContextProvider>
      <Editor {...props} />
    </TerminalContextProvider>
  </ContextProvider>
);

describe("Editor", () => {
  it("Editor renders correctly", () => {
    props = {
      consoleFocused: false,
      prompt: "$",
      commands: {},
      welcomeMessage: "",
      errorMessage: ""
    };

    const { container } = render(renderWrapper())
    expect(container).toMatchSnapshot();
  });
});
