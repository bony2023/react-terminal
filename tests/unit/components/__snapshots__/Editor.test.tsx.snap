// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Editor renders correctly 1`] = `
<div
  className="editor curvedTop null"
  style={
    Object {
      "background": "#fdf6e4",
    }
  }
>
  <span
    style={
      Object {
        "color": "#a917a8",
      }
    }
  >
    $
  </span>
  <div
    className="lineText"
  >
    <span
      className="preWhiteSpace"
    />
    <span
      className="preWhiteSpace"
    />
  </div>
</div>
`;
